crates/foundry/cheatcodes/src/inspector.rs

```diff
@@ -1449,10 +1449,48 @@ impl<
 
                 if needs_processing {
                     let mut expected_revert = std::mem::take(&mut self.expected_revert).unwrap();
+
+                    let internal_expect_revert =
+                        self.config.internal_expect_revert || expected_revert.allow_internal || {
+                            // Check if there is function-level
+                            // only if we don't have a global config allowing
+                            let identifier = ecx
+                                .journaled_state
+                                .load_account(call.target_address)
+                                .ok()
+                                .and_then(|account| account.info.code.as_ref())
+                                .and_then(|code| {
+                                    self.config
+                                        .available_artifacts
+                                        .find_by_deployed_code(code.original_byte_slice())
+                                })
+                                .and_then(|(artifact_id, contract_data)| {
+                                    let input = call.input.bytes(ecx);
+                                    (input.len() >= 4).then(|| {
+                                        let selector = input
+                                            .get(..4)
+                                            .map(Selector::from_slice)
+                                            .expect("Input must have at least 4 bytes");
+                                        contract_data
+                                            .abi
+                                            .functions()
+                                            .find(|f| f.selector() == selector)
+                                            .map(|function| TestFunctionIdentifier {
+                                                contract_artifact: artifact_id.clone(),
+                                                function_selector: function.selector().to_string(),
+                                            })
+                                    })?
+                                });
+
+                            identifier.as_ref().is_some_and(|id| {
+                                self.config.functions_internal_expect_revert.contains(id)
```

Comment:
Suggestion(non-blocking): Could this be split off into a function to reduce the amount of code, while making it more self-documenting?